src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py

```diff
@@ -237,7 +231,8 @@ def register_source_uri_arguments(scope):
 
 for item in ['file', 'blob']:
     register_cli_argument('storage {} url'.format(item), 'protocol', help='Protocol to use.', default='https', **enum_choice_list(['http', 'https']))
-    register_source_uri_arguments('storage {} copy start'.format(item))
+
+register_source_uri_arguments('storage file copy start')
```

Comment:
My recommendation would be to update the copy operation to enable full functionality for both and then merge. Or we could go ahead and merge this, but open an issue that `blob_copy` doesn't support the share -> container scenario and include it in the next PR.